fw/rbcx-coprocessor/include/Esp32Manager.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "utils/QueueWrapper.hpp"
+#include "utils/RingBuffer.hpp"
+#include "utils/TaskWrapper.hpp"
+
+// These methods are weakly linked as empty from Esp32Manager.cpp,
+// override them near the code which uses these pins.
+// * xxFreeUp - called when Esp32Manager needs to assume control of the pin,
+//              you can clean-up here if needed.
+// * xxRestore - called when Esp32Manager is done with the pins, you should call
+//               your pinInits here.
+void esp32Pin0FreeUp();
+void esp32Pin0Restore();
+void esp32Pin2FreeUp();
+void esp32Pin2Restore();
+void esp32Pin12FreeUp();
+void esp32Pin12Restore();
+void esp32Pin15FreeUp();
+void esp32Pin15Restore();
```

Comment:
Toto by možná šlo udělat nějak líp, momentálně budeme s v1.1 muset měnit i tyto callbacky, protože to, který pin ESP32 je který na STM32 se změní :/

Reply:
Ty piny se musí managovat jeden po druhém? Nemůžem to vyřešit jedním FreeUp/Restore?

Reply:
Jde, idea byla taková, že to freeUp/restore bude poblíž kódu, co ten daný pin na něco používá, viz ControlLink https://github.com/RoboticsBrno/RB3204-RBCX/pull/31/files#diff-98a8ccae468284e34a6a74cd1ab8918bR35-R44

Reply:
Když pominu budoucí debugger pro ESP, implementovaný v STM, tak ty piny budeš měnit jen na jednom místě - když přijde požadavek na reset ESP, ne?

Reply:
Ano, ale každá z pinů může v normálním stavu používat něco jiného, a momentálně to máme tak, že každá "věc" si inicializuje svoje vlastní piny.